dashboard/app/controllers/api_controller.rb

```diff
@@ -1,7 +1,39 @@
+require 'google/apis/classroom_v1'
+
 class ApiController < ApplicationController
   layout false
   include LevelsHelper
 
+  GOOGLE_AUTH_SCOPES = [
+    Google::Apis::ClassroomV1::AUTH_CLASSROOM_COURSES_READONLY,
+    Google::Apis::ClassroomV1::AUTH_CLASSROOM_ROSTERS_READONLY,
+  ].freeze
+
+  def google_classrooms
+    client = Signet::OAuth2::Client.new(
+      authorization_uri: 'https://accounts.google.com/o/oauth2/auth',
+      token_credential_uri:  'https://www.googleapis.com/oauth2/v3/token',
+      client_id: CDO.dashboard_google_key,
+      client_secret: CDO.dashboard_google_secret,
+      refresh_token: current_user.oauth_refresh_token,
+      access_token: current_user.oauth_token,
+      expires_at: current_user.oauth_token_expiration,
+      scope: GOOGLE_AUTH_SCOPES,
+    )
+    service = Google::Apis::ClassroomV1::ClassroomService.new
+    service.authorization = client
+
+    response = service.list_courses(page_size: 100)
```

Comment:
You may also want to set the teacherId to "me" and courseStates to "ACTIVE" (see https://developers.google.com/classroom/reference/rest/v1/courses/list).

I'm looking at the sample code here: https://developers.google.com/classroom/quickstart/ruby. Where does that fit in? How do you get the authorization token, which requires showing a web page to the user?